libpod/runtime_ctr.go

```diff
@@ -144,25 +141,34 @@ func (r *Runtime) removeContainer(c *Container, force bool) error {
 		return err
 	}
 
-	if err := r.state.RemoveContainer(c); err != nil {
-		return errors.Wrapf(err, "error removing container from state")
+	// Remove the container from the state
+	if c.config.Pod != "" {
+		pod, err := r.state.Pod(c.config.Pod)
+		if err != nil {
+			return errors.Wrapf(err, "container %s is in pod %s, but pod cannot be retrieved", c.ID(), pod.ID())
+		}
+
+		if err := r.state.RemoveContainerFromPod(pod, c); err != nil {
+			return errors.Wrapf(err, "error removing container %s from state", c.ID())
+		}
+	} else {
+		if err := r.state.RemoveContainer(c); err != nil {
+			return errors.Wrapf(err, "error removing container from state")
```

Comment:
Do we get enough of an err message from RemoveContainer, to just return err.

Reply:
Should be good enough, just returning error